### PR TITLE
PE-3517: Do not fail upload when USD price is unavailable

### PR DESCRIPTION
**Pull request by matibat**
--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/1ili51vmjhlko
iOS release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:ios:06ea33a95a2e0d42ffce07/releases/14106k3v1uat0

### CODE REVIEW

File: lib/blocs/upload/upload_cubit.dart [hunk 1]
```diff
@@ -293,7 +293,8 @@ class UploadCubit extends Cubit<UploadState> {
           isFreeThanksToTurbo: uploadPlan.useTurbo,
         ),
       );
-    } catch (error) {
+    } catch (error, stackTrace) {
+      print('Error while creating cost estimate: $error - $stackTrace');
```
Comment (matibat) [inline lib/blocs/upload/upload_cubit.dart:297]: Unnecessary change. Reminder to un-do this

File: lib/l10n/app_en.arb [hunk 2]
```diff
@@ -1490,6 +1490,10 @@
   "@username": {
     "description": "Label for the username"
   },
+  "usdPriceNotAvailable": "(USD price not available at the moment)",
```
Comment (matibat) [inline lib/l10n/app_en.arb:1493]: New key to be translated

File: lib/services/arweave/arweave_service.dart [hunk 3]
```diff
@@ -920,7 +920,7 @@ class ArweaveService {
     const String coinGeckoApi =
         'https://api.coingecko.com/api/v3/simple/price?ids=arweave&vs_currencies=usd';
 
-    final response = await ArDriveHTTP().getJson(coinGeckoApi);
+    final response = await ArDriveHTTP(retries: 3).getJson(coinGeckoApi);
```
Comment (matibat) [inline lib/services/arweave/arweave_service.dart:923]: Lowers the retries from 8 to 3.

File: lib/blocs/upload/cost_estimate.dart [hunk 4]
```diff
@@ -56,9 +56,21 @@ class CostEstimate {
         v2FilesPstFee.value;
 
     final arUploadCost = winstonToAr(totalCost);
-    final usdUploadCost = await arweaveService
-        .getArUsdConversionRate()
-        .then((conversionRate) => double.parse(arUploadCost) * conversionRate);
+
+    late double? usdUploadCost;
+
+    try {
```
Comment (matibat) [inline lib/blocs/upload/cost_estimate.dart:62]: The main change

**Review by matibat — COMMENTED** [review]
self-review checkpoint

File: lib/blocs/upload/upload_cubit.dart (hunk 1 above)
Comment (matibat) [reply lib/blocs/upload/upload_cubit.dart]: 37b09810c4b88ea6a1975ddca8dbd3c54403a8e2

File: lib/blocs/upload/cost_estimate.dart [hunk 5]
```diff
@@ -56,9 +56,21 @@ class CostEstimate {
...
+      if (uploadPlan.useTurbo) {
```
Comment (thiagocarvalhodev) [inline lib/blocs/upload/cost_estimate.dart:63]: Nice.

File: lib/blocs/upload/cost_estimate.dart [hunk 6]
```diff
@@ -56,9 +56,21 @@ class CostEstimate {
         v2FilesPstFee.value;
 
     final arUploadCost = winstonToAr(totalCost);
-    final usdUploadCost = await arweaveService
-        .getArUsdConversionRate()
-        .then((conversionRate) => double.parse(arUploadCost) * conversionRate);
+
+    late double? usdUploadCost;
```
Comment (thiagocarvalhodev) [inline lib/blocs/upload/cost_estimate.dart:60]: remove this `late` if the variable can be null.

File: lib/blocs/upload/cost_estimate.dart (hunk 5 above)
Comment (matibat) [reply lib/blocs/upload/cost_estimate.dart]: d7d76c925eff21fa6bee21f4b4d83fb5ece90a1c

File: lib/blocs/upload/cost_estimate.dart (hunk 6 above)
Comment (matibat) [reply lib/blocs/upload/cost_estimate.dart]: d7d76c925eff21fa6bee21f4b4d83fb5ece90a1c

**Review by thiagocarvalhodev — DISMISSED** [review]
LGTM

**Review by thiagocarvalhodev — APPROVED** [review]
LGTM!
